flocker/node/_deploy.py

```diff
@@ -681,6 +712,13 @@ def calculate_necessary_state_changes(self, local_state,
 
         :return: A ``IStateChange`` provider.
         """
+        # We are a node-specific IDeployer:
+        current_node_state = current_cluster_state.get_node(self.hostname)
+        if current_node_state.applications is None:
+            # We don't know current application state, so can't calculate
+            # anything...
```

Comment:
Maybe worth mentioning why we might not know about application state, even though we're the application deployer (ie, we don't discover stuff until dataset state has been discovered).